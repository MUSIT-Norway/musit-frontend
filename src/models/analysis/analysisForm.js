// @flow
import type { Field } from '../../forms/form';
import type { AnalysisCollection } from '../../types/analysis';
import type { CollectionId } from 'types/ids';
import type { ImportAnalysisResult } from 'types/analysisResult';

export type FormValue = {
  name: string,
  defaultValue: ?string | boolean | Array<any>
};

const toField = (
  name: string,
  defaultValue: ?string | boolean | Array<any>
): FormValue => ({
  name,
  defaultValue
});

export const fromJsonToForm: (
  json: AnalysisCollection,
  fields: Array<Field<*>>
) => Array<FormValue> = (json, formDef) => {
  const formValues = formDef.reduce(
    (acc, field) => ({
      ...acc,
      [field.name]: toField(field.name, json[field.name])
    }),
    {}
  );

  let persons = [];
  if (formValues.doneBy && formValues.doneBy.defaultValue) {
    persons = persons.concat([
      {
        name: json.doneByName,
        uuid: json.doneBy,
        role: 'doneBy',
        date: json.doneDate
      }
    ]);
  }

  if (formValues.responsible && formValues.responsible.defaultValue) {
    persons = persons.concat([
      {
        name: json.responsibleName,
        uuid: json.responsible,
        role: 'responsible',
        date: null
      }
    ]);
  }

  if (formValues.administrator && formValues.administrator.defaultValue) {
    persons = persons.concat([
      {
        name: json.administratorName,
        uuid: json.administrator,
        role: 'administrator',
        date: null
      }
    ]);
  }

  if (formValues.completedBy && formValues.completedBy.defaultValue) {
    persons = persons.concat([
      {
        name: json.completedByName,
        uuid: json.completedBy,
        role: 'completedBy',
        date: json.completedDate
      }
    ]);
  }
  formValues.persons = toField('persons', persons);

  const restriction = json.restriction;
  formValues.restrictions = toField('restrictions', !!restriction);
  if (restriction) {
    formValues.restrictions_requester = toField(
      'restrictions_requester',
      restriction.requester
    );
    formValues.restrictions_requesterName = toField(
      'restrictions_requesterName',
      restriction.requesterName
    );
    formValues.restrictions_reason = toField('restrictions_reason', restriction.reason);
    formValues.restrictions_caseNumbers = toField(
      'restrictions_caseNumbers',
      restriction.caseNumbers
    );
    formValues.restrictions_cancelledReason = toField(
      'restrictions_cancelledReason',
      restriction.cancelledReason
    );
    formValues.restrictions_expirationDate = toField(
      'restrictions_expirationDate',
      restriction.expirationDate
    );
  }

  const result = json.result;
  if (result) {
    formValues.comments = toField('comments', result.comment);
    formValues.externalSource = toField('externalSource', result.extRef);
  }

  return Object.keys(formValues).map(key => formValues[key]);
};
