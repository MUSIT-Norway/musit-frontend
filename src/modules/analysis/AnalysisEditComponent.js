/* @flow */
import React, { PropTypes } from 'react';
import { I18n } from 'react-i18nify';
import {
  Radio,
  PageHeader,
  Form,
  FormGroup,
  Col,
  FormControl,
  Button,
  Well,
  Table,
  Panel
} from 'react-bootstrap';
import FontAwesome from 'react-fontawesome';
import { SaveCancel } from '../../components/formfields/index';
import type { AppSession } from '../../types/appSession';
import type { FormData } from './types/form';
import { hashHistory } from 'react-router';
import Config from '../../config';
import Label from './components/Label';
import FieldGroup from './components/FIeldGroup';
import AddButton from './components/AddButton';
import NewLine from './components/NewLine';

type AnalysisType = { id: number, name: string };

type ObjectData = { uuid: string };

type Store = {
  objectsData: ObjectData[],
  analysisTypes: [AnalysisType]
};

type AnalysisTypeArray = [AnalysisType];

type Params = {
  analysisId?: string
};

type Props = {
  form: FormData,
  updateForm: Function,
  store: Store,
  appSession: AppSession,
  editAnalysisEvent: Function,
  params: Params
};

const getTableRow = (museumNo: ?string, subNo: ?string, term: ?string) => {
  return (
    <tr>
      <td>{museumNo}</td>
      <td>{subNo}</td>
      <td>{term}</td>
    </tr>
  );
};

const getObjectsValue = (form: FormData) => {
  if (form.type.rawValue === 'AnalysisCollection') {
    return form.events.rawValue
      ? form.events.rawValue.map(a => getTableRow(a.museumNo, a.subNo, a.term))
      : [];
  }
  return getTableRow(form.museumNo.rawValue, form.subNo.rawValue, form.term.rawValue);
};

const getValue = field => field.rawValue || '';

export const editAnalysisEventLocal = (
  appSession: AppSession,
  form: FormData,
  editAnalysisEvent: Function,
  params: Params
) =>
  editAnalysisEvent({
    id: params.analysisId,
    museumId: appSession.museumId,
    data: {
      analysisTypeId: getValue(form.analysisTypeId),
      eventDate: getValue(form.registeredDate),
      note: getValue(form.note),
      ...(form.type.rawValue === 'Analysis'
        ? { objectId: form.objectId.rawValue }
        : {
            objectIds: form.events.rawValue
              ? form.events.rawValue.map(a => a.objectId)
              : []
          }),
      restriction: form.restrictions.rawValue
        ? {
            requester: getValue(form.requester),
            expirationDate: getValue(form.expirationDate),
            reason: getValue(form.reason),
            caseNumbers: getValue(form.caseNumbers),
            cancelledBy: getValue(form.cancelledBy),
            cancelledReason: getValue(form.cancelledReason)
          }
        : null
    },
    token: appSession.accessToken
  });

const updateFormField = (field, updateForm) =>
  e =>
    updateForm({
      name: field.name,
      rawValue: e.target.value
    });

const updateFormFieldValue = (field, updateForm, value) =>
  () =>
    updateForm({
      name: field.name,
      rawValue: value
    });

const getAnalysisType = (store: Store, form: Form) => {
  const r: AnalysisTypeArray = store.analysisTypes && form.analysisTypeId
    ? store.analysisTypes.filter(a => form.analysisTypeId.value === a.id)
    : '';
  if (r.length > 0) {
    return r[0].name;
  }
  return '';
};
export const goToAnalysis = (
  fn: Promise<*>,
  appSession: AppSession,
  goTo: Function = hashHistory.push
) =>
  fn.then(updated =>
    goTo(Config.magasin.urls.client.analysis.editAnalysis(appSession, updated.id)));

const AnalysisEdit = (
  { params, form, store, updateForm, editAnalysisEvent, appSession }: Props
) => (
  <div>
    <br />
    <PageHeader style={{ paddingLeft: 20 }}>
      {I18n.t('musit.analysis.registeringAnalysis')}
    </PageHeader>
    <Col md={12}>
      <strong>Registrert:</strong>
      {' '}
      <FontAwesome name="user" />
      {' '}
      {getValue(form.registeredByName)}
      {' '}
      <FontAwesome name="clock-o" />{' '}{getValue(form.registeredDate)}
    </Col>
    <Col md={12}>
      <strong>Sist endret:</strong>
      {' '}
      <FontAwesome name="user" />
      {' '}
      {getValue(form.doneBy)}
      {' '}
      <FontAwesome name="clock-o" />
      {' '}
      {getValue(form.doneDate)}
    </Col>
    <NewLine />
    <Form horizontal style={{ paddingLeft: 10 }}>
      <FormGroup>
        <Label label="Type analyse" md={1} />
        <Col md={11}>
          {getAnalysisType(store, form)}
        </Col>
      </FormGroup>
      <FormGroup>
        <Label label="Analysested" md={1} />
        <Col md={2}>
          <FormControl componentClass="select" placeholder="Velg sted">
            <option value="Velgsted">Velg sted</option>
            <option value="other">...</option>
          </FormControl>
        </Col>
      </FormGroup>

    </Form>

    <Form>
      <FormGroup>
        <FieldGroup
          id="formControlsText"
          type="text"
          label="saksnummer"
          value={getValue(form.caseNumber)}
          onChange={updateFormField(form.caseNumber, updateForm)}
        />
      </FormGroup>
      <FormGroup>
        <AddButton id="1" label="Legg til saksnummer" md={5} />
      </FormGroup>
    </Form>
    <NewLine />
    <Form horizontal style={{ paddingLeft: 10 }}>
      <FormGroup
        controlId={form.note.name}
        validationState={form.note.status && !form.note.status.valid ? 'error' : null}
      >
        <Label label="Kommentar til analysen" md={1} />
        <Col md={5}>
          <FormControl
            className="note"
            onChange={updateFormField(form.note, updateForm)}
            componentClass="textarea"
            placeholder={form.note.name}
            value={getValue(form.note)}
          />
        </Col>
      </FormGroup>
      <FormGroup>
        <Col md={12}><h5><b>Personer tilknyttet analysen</b></h5></Col>
      </FormGroup>
      <FormGroup>
        <FieldGroup
          id="responsible"
          type="text"
          label="Navn"
          placeholder="Fornavn Etternavn"
          value={getValue(form.responsible)}
          onChange={updateFormField(form.responsible, updateForm)}
        />
        <Label label="Rolle" md={1} />
        <Col md={1}>
          <FormControl componentClass="select" placeholder="Velg rolle">
            <option value="Velgsted">Velg rolle</option>
            <option value="other">...</option>
          </FormControl>
        </Col>
        <AddButton id="3" label="Legg til person" md={2} />
      </FormGroup>
    </Form>
    <NewLine />
    <Well>
      <Form horizontal>
        <FormGroup>
          <Col md={1}>
            <h5><strong>Objekt/prøve</strong></h5>
          </Col>
        </FormGroup>
        <FormGroup>
          <Col md={11} mdOffset={1}>
            <Table responsive>
              <thead>
                <tr>
                  <th>Museumsnr</th>
                  <th>Unr</th>
                  <th>Term/artsnavn</th>
                </tr>
              </thead>
              <tbody>
                {getObjectsValue(form)}
              </tbody>
            </Table>
          </Col>
        </FormGroup>
        <FormGroup>
          <AddButton id="2" label="Legg til objekt" md={11} mdOffset={1} />
        </FormGroup>
        <FormGroup>
          <FieldGroup
            id="formControlsText"
            type="text"
            label="Ekstern kilde"
            placeholder="http://www.lenke.no"
            value={getValue(form.externalSource)}
            onChange={updateFormField(form.externalSource, updateForm)}
          />
          <Col md={2}>
            <Button>Lagre</Button>
          </Col>
        </FormGroup>
        <FormGroup>
          <FieldGroup id="formControlsText" type="text" label="Ladt opp fil" />
          <Col md={2}>
            <Button>Bla gjennom</Button>
          </Col>
        </FormGroup>
        <FormGroup>
          <Label label="Kommentar / resultat" md={1} />
          <Col md={5}>
            <FormControl
              componentClass="textarea"
              placeholder=""
              value={getValue(form.comments)}
              onChange={updateFormField(form.comments, updateForm)}
            />
          </Col>
        </FormGroup>
        <FormGroup>
          <Label label="Klausulering" md={1} />
          <Col md={5}>
            <Radio
              name="finalize"
              checked={!!form.restrictions.rawValue}
              inline
              onChange={updateFormFieldValue(form.restrictions, updateForm, true)}
            >
              Ja
            </Radio>
            <Radio
              name="finalize"
              inline
              checked={!form.restrictions.rawValue}
              onChange={updateFormFieldValue(form.restrictions, updateForm, false)}
            >
              Nei
            </Radio>
          </Col>
        </FormGroup>
        {form.restrictions.rawValue &&
          <FormGroup>
            <Panel
              collapsible
              expanded
              style={{ border: 'none', backgroundColor: '#f5f5f5' }}
            >
              <FormGroup>
                <FieldGroup
                  id="navn"
                  md={1}
                  type="text"
                  label="Klausulert for"
                  placeholder="Fornavn Etternavn"
                  value={getValue(form.requester)}
                  onChange={updateFormField(form.requester, updateForm)}
                />
              </FormGroup>
              <FormGroup>
                <FieldGroup
                  id="navn"
                  md={1}
                  type="text"
                  label="Årsak til klausulering"
                  value={getValue(form.reason)}
                  onChange={updateFormField(form.reason, updateForm)}
                />
              </FormGroup>
              <FormGroup>
                <FieldGroup
                  id="Saksnummer"
                  md={1}
                  type="text"
                  label="Saksnummer"
                  value={getValue(form.caseNumbers)}
                  onChange={updateFormField(form.caseNumbers, updateForm)}
                />
              </FormGroup>
              <FormGroup>
                <AddButton
                  id="3"
                  label="Legg til flere saksnummer"
                  md={11}
                  mdOffset={1}
                />
              </FormGroup>
              <FormGroup>
                <FieldGroup
                  id="navn"
                  md={1}
                  type="text"
                  label="Sluttdato"
                  value={getValue(form.expirationDate)}
                  onChange={updateFormField(form.expirationDate, updateForm)}
                />
              </FormGroup>
              <FormGroup>
                <FieldGroup
                  id="navn"
                  md={1}
                  type="text"
                  label="Opphevet av"
                  placeholder="Fornavn Etternavn"
                  value={getValue(form.cancelledBy)}
                  onChange={updateFormField(form.cancelledBy, updateForm)}
                />
              </FormGroup>
              <FormGroup>
                <FieldGroup
                  id="navn"
                  md={1}
                  type="text"
                  label="Årsak til oppheving"
                  placeholder=""
                  value={getValue(form.cancelledReason)}
                  onChange={updateFormField(form.cancelledReason, updateForm)}
                />
              </FormGroup>
            </Panel>
          </FormGroup>}
      </Form>
    </Well>
    <SaveCancel
      onClickSave={() =>
        goToAnalysis(
          editAnalysisEventLocal(appSession, form, editAnalysisEvent, params),
          appSession
        )}
    />
  </div>
);

const FieldShape = {
  name: PropTypes.string.isRequired,
  rawValue: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
  status: PropTypes.shape({
    valid: PropTypes.bool,
    error: PropTypes.any
  })
};

const FieldShapeBoolean = {
  name: PropTypes.string.isRequired,
  rawValue: PropTypes.bool,
  status: PropTypes.shape({
    valid: PropTypes.bool,
    error: PropTypes.any
  })
};

AnalysisEdit.propTypes = {
  form: PropTypes.shape({
    id: PropTypes.shape(FieldShape).isRequired,
    analysisTypeId: PropTypes.shape(FieldShape).isRequired,
    doneBy: PropTypes.shape(FieldShape).isRequired,
    doneDate: PropTypes.shape(FieldShape).isRequired,
    registeredBy: PropTypes.shape(FieldShape).isRequired,
    registeredDate: PropTypes.shape(FieldShape).isRequired,

    responsible: PropTypes.shape(FieldShape).isRequired,

    administrator: PropTypes.shape(FieldShape).isRequired,
    completedBy: PropTypes.shape(FieldShape).isRequired,
    completedDate: PropTypes.shape(FieldShape).isRequired,
    objectId: PropTypes.shape(FieldShape).isRequired,
    note: PropTypes.shape(FieldShape).isRequired,
    type: PropTypes.shape(FieldShape).isRequired,

    partOf: PropTypes.shape(FieldShape).isRequired,
    result: PropTypes.shape(FieldShape).isRequired,
    place: PropTypes.shape(FieldShape).isRequired,

    externalSource: PropTypes.shape(FieldShape).isRequired,
    comments: PropTypes.shape(FieldShape).isRequired,

    restrictions: PropTypes.shape(FieldShapeBoolean).isRequired,
    requester: PropTypes.shape(FieldShape).isRequired,
    expirationDate: PropTypes.shape(FieldShape).isRequired,
    reason: PropTypes.shape(FieldShape).isRequired,
    caseNumbers: PropTypes.shape(FieldShape).isRequired,
    cancelledBy: PropTypes.shape(FieldShape).isRequired,
    cancelledReason: PropTypes.shape(FieldShape).isRequired,

    completeAnalysis: PropTypes.shape(FieldShapeBoolean).isRequired,
    museumNo: PropTypes.shape(FieldShape).isRequired,
    term: PropTypes.shape(FieldShape).isRequired
  }).isRequired,
  updateForm: PropTypes.func.isRequired,
  store: PropTypes.object.isRequired,
  editAnalysisEvent: PropTypes.func
};

export default AnalysisEdit;
