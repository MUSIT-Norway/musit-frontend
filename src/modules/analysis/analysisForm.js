// @flow
import createForm, {
  getStrField,
  getArrField,
  getBoolField,
  getNumberField
} from '../../forms/form';

export const fieldsArray = [
  getNumberField('id'),
  getStrField('analysisTypeId'),
  getStrField('doneBy'),
  getStrField('doneDate'),
  getStrField('registeredBy'),
  getStrField('registeredByName'),
  getStrField('registeredDate'),
  getStrField('responsible'),
  getStrField('responsible'),
  getStrField('administrator'),
  getStrField('completedBy'),
  getStrField('completedDate'),
  getStrField('objectId'),
  getStrField('note'),
  getStrField('type'),
  getStrField('partOf'),
  getStrField('result'),
  getStrField('place'),
  getStrField('reason'),
  getStrField('externalSource'),
  getStrField('comments'),
  getBoolField('restrictions', false),
  getStrField('restrictions_requester'),
  getStrField('restrictions_expirationDate'),
  getStrField('restrictions_reason'),
  getStrField('restrictions_cancelledReason'),
  getArrField('restrictions_caseNumbers'),
  getArrField('caseNumbers'),
  getBoolField('completeAnalysis'),
  getStrField('museumNo'),
  getStrField('subNo'),
  getStrField('term'),
  getArrField('events', []),
  getStrField('analysisTypeCategory'),
  getStrField('updatedBy'),
  getStrField('updatedByName'),
  getStrField('updatedDate'),
  getNumberField('status', 1, 1, 4, 0)
];

export default createForm('analysisForm.js', fieldsArray);
