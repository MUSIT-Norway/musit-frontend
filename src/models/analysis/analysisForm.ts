// @flow
import { Field } from '../../forms/form';
import { AnalysisCollection } from '../../types/analysis';
import { Maybe, Star, MUSTFIX, BUG } from '../../types/common';
import { Person } from '../../types/person';

export type FormValue = {
  name: string;
  defaultValue: Maybe<any>;
  rawValue?: Maybe<any>;
};

const toField = (name: string, defaultValue: Maybe<any>): FormValue => ({
  name,
  defaultValue
});

export const fromJsonToForm: (
  json: AnalysisCollection,
  fields: Array<Field<Star>>
) => Array<FormValue> = (json, formDef) => {
  const formValues: MUSTFIX = formDef.reduce(
    (acc, field) => ({
      ...acc,
      [field.name]: toField(field.name, json[field.name])
    }),
    {}
  );

  let persons = [] as Person[];
  if (formValues.doneBy && formValues.doneBy.defaultValue) {
    persons = persons.concat([
      {
        name: json.doneByName as MUSTFIX,
        uuid: json.doneBy as MUSTFIX,
        role: 'doneBy',
        date: json.doneDate as MUSTFIX
      }
    ]);
  }

  if (formValues.responsible && formValues.responsible.defaultValue) {
    persons = persons.concat([
      {
        name: json.responsibleName as MUSTFIX,
        uuid: json.responsible as MUSTFIX,
        role: 'responsible',
        date: null as BUG
      }
    ]);
  }

  if (formValues.administrator && formValues.administrator.defaultValue) {
    persons = persons.concat([
      {
        name: json.administratorName as MUSTFIX,
        uuid: json.administrator as MUSTFIX,
        role: 'administrator',
        date: null as BUG
      }
    ]);
  }

  if (formValues.completedBy && formValues.completedBy.defaultValue) {
    persons = persons.concat([
      {
        name: json.completedByName as MUSTFIX,
        uuid: json.completedBy as MUSTFIX,
        role: 'completedBy',
        date: json.completedDate as MUSTFIX
      }
    ]);
  }
  formValues.persons = toField('persons', persons);

  const restriction = json.restriction;
  const hasRestriction = !!restriction && !restriction.cancelledStamp;
  formValues.restrictions = toField('restrictions', hasRestriction);
  formValues.restriction = toField('restriction', hasRestriction ? restriction : null);
  const result = json.result;
  if (result) {
    formValues.comments = toField('comments', result.comment);
    formValues.externalSource = toField('externalSource', result.extRef);
  }

  return Object.keys(formValues).map(key => formValues[key]);
};
