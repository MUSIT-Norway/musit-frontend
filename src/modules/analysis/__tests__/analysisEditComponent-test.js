import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import AnalysisEditComponent, {
  editAnalysisEventLocal,
  goToAnalysis
} from '../AnalysisEditComponent';
import { fieldsArray } from '../analysisForm';
import sinon from 'sinon';

const objectsData = [
  {
    museumNumber: '123',
    subNumber: '12345678911',
    term: 'Spyd',
    uuid: '1cbf15cb-8348-4e66-99a4-bc314da57a42'
  },
  {
    museumNumber: '124',
    subNumber: '12345678912',
    term: 'Beltering',
    uuid: '2cbf15cb-8348-4e66-99a4-bc314da57a42'
  },
  {
    museumNumber: '125',
    subNumber: '12345678913',
    term: 'Ã˜sekar',
    uuid: '3cbf15cb-8348-4e66-99a4-bc314da57a42'
  }
];

const analysisTypes = [
  {
    category: 8,
    id: '1bbf15cb-8348-4e66-99a4-bc314da57a42',
    name: '3D-skanning, laser'
  },
  {
    category: 8,
    id: 'b39399ab-aabd-4ebc-903b-adcf6876a364',
    name: '3D-skanning, strukturert lys'
  }
];

const store = {
  analysisTypes: analysisTypes,
  analysis: { ObjectId: '2cbf15cb-8348-4e66-99a4-bc314da57a42' },
  objectsData: objectsData
};

const param = {
  analysisId: 3
};

const form = fieldsArray.reduce(
  (acc, n) => ({
    ...acc,
    [n.name]: {
      name: n.name,
      rawValue: n.name === 'note' ? 'test note' : n.mapper.toRaw(n.defaultValue)
    }
  }),
  {}
);

const appSession = {
  museumId: 99,
  accessToken: '1234'
};

describe('AnalysisEditComponent', () => {
  it('editAnalysisEventLocal should call editAnalysisEvent', () => {
    const editAnalysisEvent = sinon.spy();
    const myForm = {
      analysisTypeId: {
        name: 'analysisTypeId',
        rawValue: 'b15ee459-38c9-414f-8b54-7c6439b44d3d'
      },
      registeredDate: {
        name: 'registeredDate',
        rawValue: null
      },
      note: {
        name: 'note',
        rawValue: null
      },
      by: {
        name: 'by',
        rawValue: null
      },
      expirationDate: {
        name: 'expirationDate',
        rawValue: null
      },
      caseNumbers: {
        name: 'caseNumbers',
        rawValue: null
      },
      cancelledBy: {
        name: 'cancelledBy',
        rawValue: null
      },
      cancelledReason: {
        name: 'cancelledReason',
        rawValue: null
      },
      reason: {
        name: 'reason',
        rawValue: null
      },
      events: {
        name: 'events',
        rawValue: []
      },
      type: {
        name: 'type',
        rawValue: 'AnalysisCollection' // ??
      }
    };
    editAnalysisEventLocal(appSession, myForm, editAnalysisEvent, param);
    expect(editAnalysisEvent.calledOnce).toBe(true);
    expect(editAnalysisEvent.getCall(0).args[0].museumId).toBe(99);
    expect(editAnalysisEvent.getCall(0).args[0].token).toBe('1234');
    expect(editAnalysisEvent.getCall(0).args[0].data.analysisTypeId).toBe(
      'b15ee459-38c9-414f-8b54-7c6439b44d3d'
    );
  });

  it('Call goToAnalysis.', done => {
    let url;
    const fakeGoTo = goToUrl => url = goToUrl;
    const fakeFn = new Promise(res => res({ id: 2 }));
    const fn = goToAnalysis(fakeFn, appSession, fakeGoTo);
    fn.then(() => {
      expect(url).toBe('/museum/99/collections/undefined/analysis/edit/2');
      done();
    });
  });

  it('should fire updateForm when input is changing', () => {
    const updateForm = sinon.spy();
    const wrapper = mount(
      <AnalysisEditComponent form={form} updateForm={updateForm} store={store} />
    );
    wrapper.find('.note').simulate('change', {
      target: {
        value: 'note changed'
      }
    });
    expect(updateForm.getCall(0).args[0].name).toBe('note');
    expect(updateForm.getCall(0).args[0].rawValue).toBe('note changed');
  });

  it('should render properly', () => {
    const updateForm = sinon.spy();
    const wrapper = shallow(
      <AnalysisEditComponent form={form} updateForm={updateForm} store={store} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
