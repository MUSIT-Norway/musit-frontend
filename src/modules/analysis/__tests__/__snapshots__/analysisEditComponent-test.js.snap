exports[`AnalysisEditComponent should render properly 1`] = `
<div>
  <br />
  <PageHeader
    bsClass="page-header"
    style={
      Object {
        "paddingLeft": 20
      }
    }>
    Registrer analyse
  </PageHeader>
  <Col
    bsClass="col"
    componentClass="div"
    md={12}>
    <strong>
      Registrert:
    </strong>
     
    <FontAwesome
      name="user" />
     
     
    <FontAwesome
      name="clock-o" />
     
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    md={12}>
    <strong>
      Sist endret:
    </strong>
     
    <FontAwesome
      name="user" />
     
     
    <FontAwesome
      name="clock-o" />
     
  </Col>
  <NewLine />
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={true}
    inline={false}
    style={
      Object {
        "paddingLeft": 10
      }
    }>
    <FormGroup
      bsClass="form-group">
      <LabelFormat
        label="Type analyse"
        md={1} />
      <Col
        bsClass="col"
        componentClass="div"
        md={11} />
    </FormGroup>
    <FormGroup
      bsClass="form-group">
      <LabelFormat
        label="Analysested"
        md={1} />
      <Col
        bsClass="col"
        componentClass="div"
        md={2}>
        <FormControl
          bsClass="form-control"
          componentClass="select"
          placeholder="Velg sted">
          <option
            value="Velgsted">
            Velg sted
          </option>
          <option
            value="other">
            ...
          </option>
        </FormControl>
      </Col>
    </FormGroup>
  </Form>
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={false}
    inline={false}>
    <FormGroup
      bsClass="form-group">
      <FieldGroup
        id="formControlsText"
        label="saksnummer"
        md={1}
        onChange={[Function anonymous]}
        type="text"
        value="" />
    </FormGroup>
    <FormGroup
      bsClass="form-group">
      <AddButton
        id="1"
        label="Legg til saksnummer"
        md={5}
        mdOffset={0} />
    </FormGroup>
  </Form>
  <NewLine />
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={true}
    inline={false}
    style={
      Object {
        "paddingLeft": 10
      }
    }>
    <FormGroup
      bsClass="form-group"
      controlId="note"
      validationState={null}>
      <LabelFormat
        label="Kommentar til analysen"
        md={1} />
      <Col
        bsClass="col"
        componentClass="div"
        md={5}>
        <FormControl
          bsClass="form-control"
          className="note"
          componentClass="textarea"
          onChange={[Function anonymous]}
          placeholder="note"
          value="test note" />
      </Col>
    </FormGroup>
    <FormGroup
      bsClass="form-group">
      <Col
        bsClass="col"
        componentClass="div"
        md={12}>
        <h5>
          <b>
            Personer tilknyttet analysen
          </b>
        </h5>
      </Col>
    </FormGroup>
    <FormGroup
      bsClass="form-group">
      <FieldGroup
        id="responsible"
        label="Navn"
        md={1}
        onChange={[Function anonymous]}
        placeholder="Fornavn Etternavn"
        type="text"
        value="" />
      <LabelFormat
        label="Rolle"
        md={1} />
      <Col
        bsClass="col"
        componentClass="div"
        md={1}>
        <FormControl
          bsClass="form-control"
          componentClass="select"
          placeholder="Velg rolle">
          <option
            value="Velgsted">
            Velg rolle
          </option>
          <option
            value="other">
            ...
          </option>
        </FormControl>
      </Col>
      <AddButton
        id="3"
        label="Legg til person"
        md={2}
        mdOffset={0} />
    </FormGroup>
  </Form>
  <NewLine />
  <Well
    bsClass="well">
    <Form
      bsClass="form"
      componentClass="form"
      horizontal={true}
      inline={false}>
      <FormGroup
        bsClass="form-group">
        <Col
          bsClass="col"
          componentClass="div"
          md={1}>
          <h5>
            <strong>
              Objekt/pr√∏ve
            </strong>
          </h5>
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group">
        <Col
          bsClass="col"
          componentClass="div"
          md={11}
          mdOffset={1}>
          <Table
            bordered={false}
            bsClass="table"
            condensed={false}
            hover={false}
            responsive={true}
            striped={false}>
            <thead>
              <tr>
                <th>
                  Museumsnr
                </th>
                <th>
                  Unr
                </th>
                <th>
                  Term/artsnavn
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td />
                <td />
                <td />
              </tr>
            </tbody>
          </Table>
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group">
        <AddButton
          id="2"
          label="Legg til objekt"
          md={11}
          mdOffset={1} />
      </FormGroup>
      <FormGroup
        bsClass="form-group">
        <FieldGroup
          id="formControlsText"
          label="Ekstern kilde"
          md={1}
          onChange={[Function anonymous]}
          placeholder="http://www.lenke.no"
          type="text"
          value="" />
        <Col
          bsClass="col"
          componentClass="div"
          md={2}>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}>
            Lagre
          </Button>
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group">
        <FieldGroup
          id="formControlsText"
          label="Ladt opp fil"
          md={1}
          type="text" />
        <Col
          bsClass="col"
          componentClass="div"
          md={2}>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}>
            Bla gjennom
          </Button>
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group">
        <LabelFormat
          label="Kommentar / resultat"
          md={1} />
        <Col
          bsClass="col"
          componentClass="div"
          md={5}>
          <FormControl
            bsClass="form-control"
            componentClass="textarea"
            onChange={[Function anonymous]}
            placeholder=""
            value="" />
        </Col>
      </FormGroup>
      <FormGroup
        bsClass="form-group">
        <LabelFormat
          label="Klausulering"
          md={1} />
        <Col
          bsClass="col"
          componentClass="div"
          md={5}>
          <Radio
            bsClass="radio"
            checked={false}
            disabled={false}
            inline={true}
            name="finalize"
            onChange={[Function anonymous]}>
            Ja
          </Radio>
          <Radio
            bsClass="radio"
            checked={true}
            disabled={false}
            inline={true}
            name="finalize"
            onChange={[Function anonymous]}>
            Nei
          </Radio>
        </Col>
      </FormGroup>
    </Form>
  </Well>
  <SaveCancel
    onClickSave={[Function onClickSave]} />
</div>
`;
